src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1485,16 +1485,20 @@ void RaftReplDev::on_log_found(logstore_seq_num_t lsn, log_buffer buf, void* ctx
     rreq->add_state(repl_req_state_t::LOG_FLUSHED);
     RD_LOGD("Replay log on restart, rreq=[{}]", rreq->to_string());
 
+    // 2. Pre-commit the log entry as in nuraft pre-commit was called once log appended to logstore.
+    m_listener->on_pre_commit(rreq->lsn(), rreq->header(), rreq->key(), rreq);
```

Comment:
any issue caused by this? 
for log entries with repl_lsn <= m_rd_sb->durable_commit_lsn, which means they are already definietly commited, why should we call on_precommit for them?

> Pre-commit the log entry as in nuraft pre-commit was called once log appended to logstore.

in nuraft, `on_commit` or `on_rollback` is expected after `on_precommit`. if we only call `on_precommit` for log entries with repl_lsn <= m_rd_sb->durable_commit_lsn, no later `on_commit` will be called for these log entries, not sure this is safe

Reply:
why you think there is no on_commit() be called?

Reply:
see line 1494 ，if repl_lsn > m_rd_sb->durable_commit_lsn , it will return and not call handle _commit

Reply:
sorry , you are right, i misread line 1494